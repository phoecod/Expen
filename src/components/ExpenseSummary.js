import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {

	const expenseWord = expensesCount > 1 ? 'expenses' : 'expense';
	const totalExpensesFormat = numeral(expensesTotal).format('$0.00');
	return (
		<div>
				{
					<h1> 
						Viewing {expensesCount} {expenseWord} totalling {totalExpensesFormat}
					</h1>
				}
		</div>
	);
}

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expensesTotal: selectExpensesTotal(visibleExpenses),
		expensesCount: visibleExpenses.length
	}
}

export default connect(mapStateToProps)(ExpenseSummary);
