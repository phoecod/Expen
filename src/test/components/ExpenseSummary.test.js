import React from 'react';
import {shallow} from 'enzyme';
import ExpenseSummary from '../../components/ExpenseSummary';


test('display summary expense page', ()=> {
	const wrapper = shallow(<ExpenseSummary expensesCount={3} expensesTotal={200}/>);
	expect(wrapper).toMatchSnapshot();
});