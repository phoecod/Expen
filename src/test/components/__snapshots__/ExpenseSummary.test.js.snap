// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display summary expense page 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $200.00
  </h1>
</div>
`;
